### PR TITLE
Warn init coords

### PR DESCRIPTION
**Pull request by fgrunewald**
Raise warning if provided meta-mol coordinates are out-of bounds

### CODE REVIEW

File: polyply/src/linalg_functions.py [hunk 1]
```diff
@@ -235,3 +235,38 @@ def _rotate_xyz(object_xyz, theta_x, theta_y, theta_z):
     return rotated_object
 
 rotate_xyz = jit(_rotate_xyz)
+
+def not_exceeds_max_dimensions(point, maxdim):
+    """
+    Check if point is within the compontents of
+    the maxdim vector. Note that all coodinates in
+    polyply are definiet positive.
```
Comment (pckroon) [inline polyply/src/linalg_functions.py:243]: typo
Also semi-positive definite, since 0 is also fine.

File: polyply/src/linalg_functions.py [hunk 2]
```diff
@@ -235,3 +235,38 @@ def _rotate_xyz(object_xyz, theta_x, theta_y, theta_z):
...
+    Parameters:
+    -----------
+    point: np.ndarray
+    maxdim: np.ndarray
+
+    Returns:
+    --------
+    bool
+    """
+    return np.all(point <= maxdim) and np.all(point >= np.array([0., 0., 0.]))
```
Comment (pckroon) [inline polyply/src/linalg_functions.py:254]:
```suggestion
    return np.all(0 <= point < maxdim))
```
I'm just not 100% sure this works with np arrays

**Review by pckroon — CHANGES_REQUESTED** [review]

File: polyply/src/linalg_functions.py (hunk 1 above)
Comment (fgrunewald) [reply polyply/src/linalg_functions.py:243]: good catch didn't update the doc-string

File: polyply/src/linalg_functions.py (hunk 2 above)
Comment (fgrunewald) [reply polyply/src/linalg_functions.py:254]: doesn't work for me. As it's written now is not super neat but functional. To be honest it wasn't worth figuring out how to numpyfy it.

File: polyply/src/nonbond_engine.py [hunk 3]
```diff
@@ -318,7 +319,16 @@ def from_topology(cls, molecules, topology, box):
...
+                    # check if position is inside grid
+                    if not_exceeds_max_dimensions(molecule.nodes[node]["position"], box):
+                        positions[idx, :] = molecule.nodes[node]["position"]
+                    else:
+                        mol_name = molecule.mol_name
+                        resname = molecule.nodes[node]['resname']
+                        msg = (f"Provided coordinate for residue {resname} in molecule {mol_name}\n"
+                               f"with molecule index {idx} exceeds maximum box dimensions.\n"
+                                "Make sure all coordiantes are wrapped inside the box.")
+                        raise IOError(msg)
```
[the remark this answers is not in the document]
Comment (fgrunewald) [reply polyply/src/nonbond_engine.py:331]: yep only once per molecule only one time in the code; so I think this way is fine. It's also not the performance bottleneck. I'll add the check on nan's that's a good idea

File: polyply/src/linalg_functions.py [hunk 4]
```diff
@@ -235,3 +235,38 @@ def _rotate_xyz(object_xyz, theta_x, theta_y, theta_z):
...
+
+def pbc_complete(point, maxdim):
+    """
+    Wrap point around pbc conditions to keep
+    points from being larger than the compontents of
+    the maxdim vector. Note that all coodinates in
+    polyply are definiet positive.
```
Comment (pckroon) [inline polyply/src/linalg_functions.py:261]: Still typo :) 
Also semi-definite

File: polyply/src/nonbond_engine.py [hunk 5]
```diff
@@ -318,7 +319,24 @@ def from_topology(cls, molecules, topology, box):
...
+                        msg = (f"Provided coordinate for residue {resname} in molecule {mol_name}\n"
+                               f"with molecule index {idx} is not finite. All coordinates in\n"
+                                "polyply must be positive semi-definite.")
+                        raise IOError(msg)
+                    else:
+                        mol_name = molecule.mol_name
+                        resname = molecule.nodes[node]['resname']
+                        msg = (f"Provided coordinate for residue {resname} in molecule {mol_name}\n"
+                               f"with molecule index {idx} exceeds maximum box dimensions.\n"
+                                "Make sure all coordiantes are wrapped inside the box.")
```
Comment (pckroon) [inline polyply/src/nonbond_engine.py:338]: I personally wouldn't put the \n in the messages. Let thelogging module figure out formatting

**Review by pckroon — CHANGES_REQUESTED** [review]

File: polyply/src/nonbond_engine.py (hunk 5 above)
Comment (fgrunewald) [reply polyply/src/nonbond_engine.py:338]: well my experience is that the logger just dumps it into one line, which users find hard to decipher. A bit of line wrapping goes a long way.

Comment (pckroon) [reply polyply/src/nonbond_engine.py:338]: Mjeh. The cleanest solution would be to make/adapt a logging formatter.

**Review by pckroon — APPROVED** [review]

**Pull request merged**